### PR TITLE
Shorten docstrings by removing unneeded :class:/:func: + rewordings.

### PR DESCRIPTION
**Pull request by anntzer**
## PR Checklist

- [ ] Has Pytest style unit tests
- [ ] Code is [Flake 8](http://flake8.pycqa.org/en/latest/) compliant
- [ ] New features are documented, with examples if plot related
- [ ] Documentation is sphinx and numpydoc compliant
- [ ] Added an entry to doc/users/next_whats_new/ if major new feature (follow instructions in README.rst there)
- [ ] Documented in doc/api/api_changes.rst if API changed in a backward-incompatible way

### CODE REVIEW

File: lib/matplotlib/transforms.py [hunk 1]
```diff
@@ -761,25 +758,18 @@ def invalidate(self):
 
     @staticmethod
     def unit():
-        """
-        (staticmethod) Create a new unit :class:`Bbox` from (0, 0) to
```
Comment (anntzer) [inline lib/matplotlib/transforms.py]: (both sphinx and pydoc already mark the method as static)

File: lib/matplotlib/axes/_axes.py [hunk 2]
```diff
@@ -5967,10 +5964,9 @@ def pcolormesh(self, *args, alpha=None, norm=None, cmap=None, vmin=None,
 
         .. note::
 
-           ``pcolormesh()`` is similar to :func:`~Axes.pcolor`. It's much
-           faster and preferred in most cases. For a detailed discussion on
-           the differences see
-           :ref:`Differences between pcolor() and pcolormesh()
+           `~Axes.pcolormesh()` is similar to `~Axes.pcolor`. It's much faster
```
Comment (timhoffm) [inline lib/matplotlib/axes/_axes.py]:
```suggestion
           `~Axes.pcolormesh` is similar to `~Axes.pcolor`. It's much faster
```

File: lib/matplotlib/transforms.py [hunk 3]
```diff
@@ -2693,8 +2669,10 @@ class TransformedPath(TransformNode):
     """
     def __init__(self, path, transform):
         """
-        Create a new :class:`TransformedPath` from the given
-        :class:`~matplotlib.path.Path` and :class:`Transform`.
+        Parameters
+        ----------
+        path : `~.path.Path`
+        transform : Transform
```
Comment (timhoffm) [inline lib/matplotlib/transforms.py]: Is Transform intentionally not linked?

**Comment by anntzer** [comment]
both points fixed, thanks for the review.

**Pull request merged**